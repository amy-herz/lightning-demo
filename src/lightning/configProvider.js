import { getDefaultConfig } from './defaultConfig';

// Closure to hold the APIs if and when available
let PROVIDED_IMPL = getDefaultConfig();
let FORM_FACTOR;

let configProvided = false;

export default function configProviderService(serviceAPI) {
  if (!configProvided) {
    PROVIDED_IMPL = {
      getCoreInfo:
        serviceAPI.getInitializer &&
        serviceAPI.getInitializer('coreInfoConfig'),
      getPathPrefix: serviceAPI.getPathPrefix,
      getFormFactor: serviceAPI.getFormFactor,
      getToken: serviceAPI.getToken,
      getLocale: serviceAPI.getLocale,
      getLocalizationService: serviceAPI.getLocalizationService,
      sanitizeDOM: serviceAPI.sanitizeDOM
    };
    configProvided = true;
    FORM_FACTOR = undefined;
  } else {
    throw new Error(
      'ConfigProvider can only be set once at initilization time'
    );
  }

  return { name: 'lightning-config-provider' };
}

export function getPathPrefix() {
  return (
    (PROVIDED_IMPL &&
      PROVIDED_IMPL.getPathPrefix &&
      PROVIDED_IMPL.getPathPrefix()) ||
    ''
  );
}

export function getToken(name) {
  return (
    PROVIDED_IMPL && PROVIDED_IMPL.getToken && PROVIDED_IMPL.getToken(name)
  );
}

export function getLocale() {
  return PROVIDED_IMPL && PROVIDED_IMPL.getLocale && PROVIDED_IMPL.getLocale();
}

export function getFormFactor() {
  if (!FORM_FACTOR) {
    if (PROVIDED_IMPL && PROVIDED_IMPL.getFormFactor) {
      FORM_FACTOR = PROVIDED_IMPL.getFormFactor();
    } else {
      FORM_FACTOR = 'DESKTOP';
    }
  }

  return FORM_FACTOR;
}

export function getLocalizationService() {
  return (
    PROVIDED_IMPL &&
    PROVIDED_IMPL.getLocalizationService &&
    PROVIDED_IMPL.getLocalizationService()
  );
}

export function sanitizeDOM(dirty, config) {
  if (PROVIDED_IMPL && PROVIDED_IMPL.sanitizeDOM) {
    return PROVIDED_IMPL.sanitizeDOM(dirty, config);
  }
  return '';
}

export function getCoreInfo() {
  return (
    (PROVIDED_IMPL && PROVIDED_IMPL.getCoreInfo) || {
      untrustedContentDomain: '.a.forceusercontent.com',
      localhostPort: '',
      securePort: ''
    }
  );
}
