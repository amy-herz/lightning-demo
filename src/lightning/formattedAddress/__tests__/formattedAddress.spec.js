import { createElement } from 'lwc';
import Element from 'lightning/formattedAddress';
// eslint-disable-next-line lwc/no-aura-libs
import { utils as localeUtils } from 'lightning:IntlLibrary';

const defaultAttributes = {
    street: '121 Spear St.',
    city: 'San Francisco',
    country: 'US',
    province: 'CA',
    postalCode: '94105',
};

const createFormattedAddress = params => {
    const element = createElement('lightning-formatted-address', {
        is: Element,
    });
    Object.assign(element, defaultAttributes, params);
    document.body.appendChild(element);
    return element;
};

function setLocaleMock(locale) {
    localeUtils.getLocaleTag = () => locale;
}

describe('formatted-address', () => {
    setLocaleMock('cn-CN');

    it('format address based on locale', () => {
        const element = createFormattedAddress();
        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    it('disabled formatted-address has no link', () => {
        const element = createFormattedAddress({
            disabled: true,
        });
        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    it('uses latitude and longitude for title and url if both exist', () => {
        const element = createFormattedAddress({
            latitude: '123',
            longitude: '456',
        });
        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    it('does not use latitude and longitude for title and url if latitude does not exist', () => {
        const element = createFormattedAddress({
            longitude: '456',
        });
        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    it('does not use latitude and longitude for title and url if longitude does not exist', () => {
        const element = createFormattedAddress({
            latitude: '123',
        });
        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });
});
