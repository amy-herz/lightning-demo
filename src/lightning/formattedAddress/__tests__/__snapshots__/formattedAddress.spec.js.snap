// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatted-address disabled formatted-address has no link 1`] = `
<lightning-formatted-address>
  #shadow-root(open)
    <div
      class="slds-truncate"
    >
      US
    </div>
    <div
      class="slds-truncate"
    >
      CA, San Francisco
    </div>
    <div
      class="slds-truncate"
    >
      121 Spear St., 94105
    </div>
</lightning-formatted-address>
`;

exports[`formatted-address does not use latitude and longitude for title and url if latitude does not exist 1`] = `
<lightning-formatted-address>
  #shadow-root(open)
    <a
      href="https://www.google.com/maps/?q=US%0ACA,%20San%20Francisco%0A121%20Spear%20St.,%2094105"
      rel="noopener"
      target="_blank"
      title="US
CA, San Francisco
121 Spear St., 94105"
    >
      <div
        class="slds-truncate"
      >
        US
      </div>
      <div
        class="slds-truncate"
      >
        CA, San Francisco
      </div>
      <div
        class="slds-truncate"
      >
        121 Spear St., 94105
      </div>
    </a>
</lightning-formatted-address>
`;

exports[`formatted-address does not use latitude and longitude for title and url if longitude does not exist 1`] = `
<lightning-formatted-address>
  #shadow-root(open)
    <a
      href="https://www.google.com/maps/?q=US%0ACA,%20San%20Francisco%0A121%20Spear%20St.,%2094105"
      rel="noopener"
      target="_blank"
      title="US
CA, San Francisco
121 Spear St., 94105"
    >
      <div
        class="slds-truncate"
      >
        US
      </div>
      <div
        class="slds-truncate"
      >
        CA, San Francisco
      </div>
      <div
        class="slds-truncate"
      >
        121 Spear St., 94105
      </div>
    </a>
</lightning-formatted-address>
`;

exports[`formatted-address format address based on locale 1`] = `
<lightning-formatted-address>
  #shadow-root(open)
    <a
      href="https://www.google.com/maps/?q=US%0ACA,%20San%20Francisco%0A121%20Spear%20St.,%2094105"
      rel="noopener"
      target="_blank"
      title="US
CA, San Francisco
121 Spear St., 94105"
    >
      <div
        class="slds-truncate"
      >
        US
      </div>
      <div
        class="slds-truncate"
      >
        CA, San Francisco
      </div>
      <div
        class="slds-truncate"
      >
        121 Spear St., 94105
      </div>
    </a>
</lightning-formatted-address>
`;

exports[`formatted-address uses latitude and longitude for title and url if both exist 1`] = `
<lightning-formatted-address>
  #shadow-root(open)
    <a
      href="https://www.google.com/maps/?q=123,456"
      rel="noopener"
      target="_blank"
      title="123,456"
    >
      <div
        class="slds-truncate"
      >
        US
      </div>
      <div
        class="slds-truncate"
      >
        CA, San Francisco
      </div>
      <div
        class="slds-truncate"
      >
        121 Spear St., 94105
      </div>
    </a>
</lightning-formatted-address>
`;
